front/lib/run_app.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  cloneBaseConfig,
+  DustProdActionRegistry,
+} from "@app/lib/actions_registry";
+import { planForWorkspace, prodAPICredentialsForOwner } from "@app/lib/auth";
+import { DustAPI } from "@app/lib/dust_api";
+import { Workspace } from "@app/lib/models";
+import { WorkspaceType } from "@app/types/user";
+
+export async function runAppOnWorkspaceId(
+  workspaceId: string,
+  appName: string,
+  inputs: any
+) {
+  const owner = await _getSystemOwnerForWorkspaceId(workspaceId);
+  const prodCredentials = await prodAPICredentialsForOwner(owner);
+  const prodAPI = new DustAPI(prodCredentials);
+  const app = cloneBaseConfig(DustProdActionRegistry[appName]?.app);
+  const config = cloneBaseConfig(DustProdActionRegistry[appName]?.config);
+
+  return await prodAPI.runApp(app, config, inputs);
+}
+
+async function _getSystemOwnerForWorkspaceId(
```

Comment:
As per above to be killed with fire.

Reason: it's part of the authorization layer which is provided by the methods in `auth.ts` and generally result in an `owner: WorkspaceType` object

Reply:
File killed with fire: 
-> function to get a system owner from a workspace id renamed and moved to auth.